### PR TITLE
Merge replay

### PR DESCRIPTION
**Pull request by danielballan**
This pulls the entire history of replay into the databroker repo and places it in `databroker/databroker/replay`, alongside `broker/` and `muggler/`.

I combined the projects' `README`, `.gitignore`, `setup.py`, and `run_tests.py`. I adapted replay's conda recipe for databroker, which had none. All of these changes should be reviewed.

### CODE REVIEW

**Comment by danielballan** [comment]
New commits remove my copy/pasted version of `DataMuggler` and correctly "git mv" it from replay, leaving authorship intact. My subsequent edits to the code are cherry-picked on top. In summary, nothing is changed, but the history is better.

File: setup.py [hunk 1]
```diff
@@ -1,16 +1,81 @@
...
               'databroker.server', 'databroker.commands',
-              'databroker.broker', 'databroker.muggler'
+              'databroker.broker', 'databroker.muggler',
               ],
     include_dirs=[np.get_include()],
-    )
+    entry_points={
+        'console_scripts': [
+        'replay = replay.run:main']},
+    package_data={'replay.gui': ['*.enaml']},
```
Comment (tacaswell) [inline setup.py]: This should be 'broker.replay.gui'

File: setup.py [hunk 2]
```diff
@@ -1,16 +1,81 @@
...
+              'databroker.replay.model',
               'databroker.server', 'databroker.commands',
-              'databroker.broker', 'databroker.muggler'
+              'databroker.broker', 'databroker.muggler',
               ],
     include_dirs=[np.get_include()],
-    )
+    entry_points={
+        'console_scripts': [
+        'replay = replay.run:main']},
```
Comment (tacaswell) [inline setup.py]: This path needs to be fixed up too

**Comment by danielballan** [comment]
There are still a _ton_ of broken imports like `from replay ...`.

**Comment by tacaswell** [comment]
@ericdill Do we want to merge this as-is and fixup replay as we go or fix it up frist?

I am inclined to fix it as we go due to the brokenness (?) that I expect to come come from mds/ophyd changes.

**Comment by danielballan** [comment]
I believe this fixes all the imports, but I can't properly test it myself. In a surprise to no one, I agree that we should fix as we go in this case. :- )

I want to build muggler on top of this....

**Pull request merged**